PVRTexLibNET/PVRTexLibNET.cs

```diff
@@ -92,49 +92,102 @@ public enum CompressorQuality
-            gcHandle.Free();
-        }
-
+        public static extern IntPtr CreateTexture(IntPtr data, uint u32Width, uint u32Height, uint u32Depth, PixelFormat ptFormat, bool preMultiplied, VariableType eChannelType, ColourSpace eColourspace);
+        
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void CreateTexture(string filePath);
+        public static extern IntPtr CreateTexture(string filePath);
 
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern bool SaveTexture(string filePath);
+        public static extern bool SaveTexture(IntPtr pPvrTexture, string filePath);
 
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void DestroyTexture();
+        public static extern void DestroyTexture(IntPtr pPvrTexture);
 
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern bool Resize(uint u32NewWidth, uint u32NewHeight, uint u32NewDepth, ResizeMode eResizeMode);
+        public static extern bool Resize(IntPtr pPvrTexture, uint u32NewWidth, uint u32NewHeight, uint u32NewDepth, ResizeMode eResizeMode);
 
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern bool GenerateMIPMaps(ResizeMode eFilterMode, uint uiMIPMapsToDo = int.MaxValue);
+        public static extern bool GenerateMIPMaps(IntPtr pPvrTexture, ResizeMode eFilterMode, uint uiMIPMapsToDo = int.MaxValue);
 
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern bool Transcode(PixelFormat ptFormat, VariableType eChannelType, ColourSpace eColourspace, CompressorQuality eQuality = CompressorQuality.PVRTCNormal, bool bDoDither = false);
+        public static extern bool Transcode(IntPtr pPvrTexture, PixelFormat ptFormat, VariableType eChannelType, ColourSpace eColourspace, CompressorQuality eQuality = CompressorQuality.PVRTCNormal, bool bDoDither = false);
 
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern uint GetTextureDataSize(int iMIPLevel = -1);
+        public static extern uint GetTextureDataSize(IntPtr pPvrTexture, int iMIPLevel = -1);
 
         [DllImport(dllName, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void GetTextureData(IntPtr data, uint dataSize, uint uiMIPLevel = 0);
+        public static extern void GetTextureData(IntPtr pPvrTexture, IntPtr data, uint dataSize, uint uiMIPLevel = 0);
+        #endregion Interop
+
+        private IntPtr _pPvrTexture = IntPtr.Zero;
 
-        public static void GetTextureData<T>(T[] data, uint dataSize, uint uiMIPLevel = 0)
+        public IntPtr PvrTexturePointer { get { return _pPvrTexture; } }
+        
+        public PVRTexture(string filePath)
+        {
+            IntPtr pPvrTexture = CreateTexture(filePath);
```

Comment:
This appears wrong.  Assigning the PVRTexture to a local variable, but then doing nothing with it.